Assets/MRTK/SDK/Features/UX/Scripts/InteractiveElement/States/StateManager.cs

```diff
@@ -0,0 +1,422 @@
+                }
+            }
+            else
+            {
+                Debug.LogError($"The {stateName} state is not being tracked and was not removed.");
+            }
+        }
+
+        /// <summary>
+        /// Removes a Core Interaction State. The state will no longer be tracked if it is removed.
+        /// </summary>
+        /// <param name="coreState">The Core Interaction State to remove</param>
+        public void RemoveState(CoreInteractionState coreState)
+        {
+            RemoveState(coreState.ToString());
+        }
+
+        /// <summary>
+        /// Check if a state is currently active.
+        /// </summary>
+        /// <param name="stateName">The name of the state to check</param>
+        /// <returns>True if the state is active, false if the state is not active</returns>
+        public bool IsStateActive(string stateName)
+        {
+            InteractionState state = GetState(stateName);
+
+            if (state != null)
+            {
+                return state.Active == 1;
+            }
+            else
+            {
+                Debug.LogError($"The {stateName} state is not being tracked, add this state using AddNewState(state) to track whether or not it is active.");
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Check if a core state is currently active.
+        /// </summary>
+        /// <param name="coreState">The name of the core state to check</param>
+        /// <returns>True if the state is active, false if the state is not active</returns>
+        public bool IsStateActive(CoreInteractionState coreState)
+        {
+            return IsStateActive(coreState.ToString());
+        }
+
+
+        /// <summary>
+        /// Check if a state is currently being tracked.
+        /// </summary>
+        /// <param name="stateName">The name of the state to check</param>
+        /// <returns>True if the state is being tracked, false if the state is not being tracked</returns>
+        public bool IsStateTracking(string stateName)
+        {
+            InteractionState state = GetState(stateName);
+
+            return state != null;
+        }
+
+        /// <summary>
+        /// Check if a state is currently being tracked.
+        /// </summary>
+        /// <param name="coreState">The name of the core state to check</param>
+        /// <returns>True if the state is being tracked, false if the state is not being tracked</returns>
+        public bool IsStateTracking(CoreInteractionState coreState)
+        {
+            return IsStateTracking(coreState.ToString());
+        }
+
+        /// <summary>
+        /// Creates and adds a new state to track given the new state name.
+        /// </summary>
+        /// <param name="stateName">The name of the state to add</param>
+        /// <returns>The new state added</returns>
+        public InteractionState AddNewState(string stateName)
```

Comment:
Why doesn't adding a new user-defined state also create an event configuration for it? Does user need to do this manually? How would I do this from code?

Reply:
A user would need to create an event configuration manually right now, but I think the goal should be: 

```
// When a new state is created, an event configuration for the new state is also created.
interactiveElement.AddNewState("Keyboard").

var keyboardStateEvents = interactiveElement.GetStateEvents<StateEvents>("Keyboard");

keyboardStateEvents.OnStateOn.AddListener(...);

keyboaradStateEvents.OnStateOff.AddListener(...)
```

@julenka does the example code above demonstrate how you would expect to access state events of a newly defined state?

Reply:
Yup, resolving. It will be good to put this in the docs somewhere.